tests/karate/src/test/java/common/net/MultiMsgWebSocketClient.java

```diff
@@ -64,11 +64,13 @@ public String getBackendResponseWithBroadcast(){
     String answer1 = getBuffer().takeTimeout(5000);
     String answer2 = getBuffer().takeTimeout(5000);
     String result = answer1.contains("result") ? answer1 : answer2;
+    assert (result.contains("\"id\":" + idAssociatedWithSentMessage));
```

Comment:
Beware, in the json specification, nothing says that a json should have the form `"id":12`, it could as well be `"id" : 13` and your code would break.

This is not especially important in this case, but the best practice is to convert the string to json and do assertions on that object.

Reply:
Agree